kahuna/public/js/components/gr-panel-button/gr-panel-button.js

```diff
@@ -0,0 +1,33 @@
+import angular from 'angular';
+import template from './gr-panel-button.html!text';
+import '../../util/rx';
+
+export const panelButton = angular.module('gr.panelButton', ['util.rx']);
+
+panelButton.controller('GrPanelButton', ['$scope', 'inject$', 'subscribe$', function($scope, inject$, subscribe$) {
+
+    const ctrl = this;
+    subscribe$($scope, ctrl.panel$, panel => {
+        ctrl.hidePanel   = () => panel.setHidden(true);
+        ctrl.showPanel   = () => panel.setHidden(false);
+        ctrl.lockPanel   = () => panel.setLocked(true);
+        ctrl.unlockPanel = () => panel.setLocked(false);
+
+        // TODO: Could we have a helper to watch multiple streams?
+        inject$($scope, panel.hidden$, ctrl, 'hidden');
+        inject$($scope, panel.locked$, ctrl, 'locked');
+    })
+}]);
+
+panelButton.directive('grPanelButton', [function() {
+    return {
+        restrict: 'E',
+        template: template,
+        bindToController: true,
+        controller: 'GrPanelButton',
+        controllerAs: 'ctrl',
+        scope: {
+            panel$: '=grPanel'
```

Comment:
Multiple.
It's a way of avoiding a client calling `.getPanel` if it hasn't been created somewhere.

[I've added a note here](https://github.com/guardian/grid/pull/1566/files#diff-7965c78e2f73afa5a99c770c34b37dedR8).

There were two other options (with their problems):
* A promise, although I didn't know what the fail would be.
* Have them as resolves in the view - but that felt like we were coupling the panels to something not really related i.e. the panel controller wasn't really controlling itself.